__tests__/cron/validateActiveUsers.ts

```diff
@@ -0,0 +1,249 @@
+      {
+        action: 'destroy',
+        type: 'person',
+        identifiers: { id: '3' },
+      },
+      {
+        action: 'destroy',
+        type: 'person',
+        identifiers: { id: '5' },
+      },
+    ];
+
+    expect(postSpy).toHaveBeenCalledWith('/users', { batch });
+
+    const fromRemovalOnly = postSpy.mock.calls[0][1].batch.every(
+      ({ identifiers }) => ['3', '5'].includes(identifiers.id),
+    );
+    expect(fromRemovalOnly).toBeTruthy();
+
+    const unRegisteredOnly = postSpy.mock.calls[0][1].batch.every(
+      ({ identifiers }) => !['vordr'].includes(identifiers.id),
+    );
+    expect(unRegisteredOnly).toBeTruthy();
+
+    const unregistered = await con
+      .getRepository(User)
+      .findOne({ select: ['cioRegistered'], where: { id: '3' } });
+    expect(unregistered.cioRegistered).toEqual(false);
+
+    const removed = await con.getRepository(UserPersonalizedDigest).count();
+    expect(removed).toBeLessThan(digests);
+  });
+});
+
+describe('users for reactivation', () => {
+  it('should not do anything if users are registered to CIO already', async () => {
+    jest.spyOn(gcp, 'getUsersActiveState').mockResolvedValue({
+      reactivateUsers: ['1', '2'],
+      inactiveUsers: ['3'],
+      downgradeUsers: ['4'],
+    });
+    const postSpy = jest
+      .spyOn(cioModule.cioV2.request, 'post')
+      .mockResolvedValue({});
+
+    // to stop running removal of `inactiveUsers`
+    await con.getRepository(User).update({ id: '3' }, { cioRegistered: false });
+
+    // default value for `cioRegistered` is true
+    await expectSuccessfulCron(cron);
+
+    expect(postSpy).not.toHaveBeenCalled();
+  });
+
+  it('should send reactivation to cio', async () => {
+    jest.spyOn(gcp, 'getUsersActiveState').mockResolvedValue({
+      reactivateUsers: ['1', '2'],
+      inactiveUsers: ['3'],
+      downgradeUsers: ['4'],
+    });
+
+    const postSpy = jest
+      .spyOn(cioModule.cioV2.request, 'post')
+      .mockResolvedValue({});
+
+    // to stop running removal of `inactiveUsers`
+    await con
+      .getRepository(User)
+      .update({ id: In(['3', '1']) }, { cioRegistered: false });
+
+    // default value for `cioRegistered` is true
+    await expectSuccessfulCron(cron);
+
+    const batch = [
+      {
+        action: 'identify',
+        type: 'person',
+        identifiers: { id: '1' },
+        attributes: {
+          accepted_marketing: false,
+          'cio_subscription_preferences.topics.topic_4': false,
+          'cio_subscription_preferences.topics.topic_7': true,
+          'cio_subscription_preferences.topics.topic_8': true,
+          'cio_subscription_preferences.topics.topic_9': true,
+          created_at: 1656427727,
+          first_name: 'Ido',
+          name: 'Ido',
+          permalink: 'http://localhost:5002/idoshamun',
+          referral_link: 'http://localhost:5002/join?cid=generic&userid=1',
+          updated_at: undefined,
+          username: 'idoshamun',
```

Comment:
Notice: these are the only attributes we are now sending to CIO for reactivation. Feel free to comment if it is missing any other properties.